### PR TITLE
:book: Release cycle leads & shadow name added

### PR DESCRIPTION
**Pull request by sayantani11**
**What this PR does / why we need it**:
Adds release lead and shadow names for v1.4.0

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by oscr** [comment]
/retitle :book: Release cycle leads & shadow name added

**Comment by oscr** [comment]
@sayantani11 It's usually a requirement to squash commits as described in https://github.com/kubernetes-sigs/cluster-api/blob/main/CONTRIBUTING.md#contributing-a-patch

**Comment by oscr** [comment]
lgtm pending the squash

File: docs/developer/release-cycle.md [hunk 1]
```diff
@@ -108,6 +108,6 @@ style: [Kubernetes 1.26 Release Team](https://github.com/kubernetes/sig-release/
 
 | **Role**                                  | **Lead Name** (**GitHub / Slack ID**) | **Shadow Name(s) (GitHub / Slack ID)** |
 |-------------------------------------------|---------------------------------------|----------------------------------------|
-| Release Lead                              |                                       |                                        |
-| Communications/Docs/Release Notes Manager |                                       |                                        |
-| CI Signal/Bug Triage/Automation Manager   |                                       |                                        |
+| Release Lead                              |  Yuvraj Kakaraparthi([@ykakarap](https://github.com/ykakarap)/Slack:`@Yuvaraj Kakaraparthi`)  | Jim Ntosas([@dntosas](https://github.com/dntosas)/ Slack:`@dntosas`), Carlos Panato([cpanato](https://github.com/cpanato)/ Slack:`@cpanato`), Amit Kumar([@hackeramitkumar](https://github.com/hackeramitkumar)/ Slack:`@Amit Kumar`) |
```
Comment (furkatgofurov7) [inline docs/developer/release-cycle.md:111]: By comparing the Slack ID, seems the Release lead name is incorrect? @ykakarap might confirm

Comment (oscr) [reply docs/developer/release-cycle.md:111]: Great catch! :bow:

Comment (ykakarap) [inline docs/developer/release-cycle.md:111]:
```suggestion
| Release Lead                              |  Yuvaraj Kakaraparthi([@ykakarap](https://github.com/ykakarap)/Slack:`@Yuvaraj Kakaraparthi`)  | Jim Ntosas([@dntosas](https://github.com/dntosas)/ Slack:`@dntosas`), Carlos Panato([cpanato](https://github.com/cpanato)/ Slack:`@cpanato`), Amit Kumar([@hackeramitkumar](https://github.com/hackeramitkumar)/ Slack:`@Amit Kumar`) |
```

Comment (ykakarap) [reply docs/developer/release-cycle.md:111]: Thanks for catching this!

**Comment by sayantani11** [comment]
tide/merge-method-squash

**Comment by sbueringer** [comment]
@sayantani11 using tide squash will lead to the PR not showing up in our release notes

**Comment by sayantani11** [comment]
> @sayantani11 using tide squash will lead to the PR not showing up in our release notes

I thought this won't be a issue since the release note is NONE. But then I'll push with squashing the commits then.

**Comment by sbueringer** [comment]
We are using a different release note generation tool than e.g. k/k. It generates release notes based on PR tiltes (including the icons) and the merge commits.

**Comment by fabriziopandini** [comment]
@ykakarap PTAL

**Pull request closed without merging**